src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -211,9 +211,63 @@ public void PeerDisconnected(int peerId)
         }
 
         /// <inheritdoc />
-        public Task<ChainedHeaderBlock> BlockMined(Block block)
+        public async Task<ChainedHeaderBlock> BlockMinedAsync(Block block)
         {
-            throw new NotImplementedException();
+            this.logger.LogTrace("({0}:{1})", nameof(block), block.GetHash());
+
+            PartialValidationResult partialValidationResult;
+
+            using (await this.reorgLock.LockAsync().ConfigureAwait(false))
+            {
+                ChainedHeader chainedHeader;
+
+                lock (this.peerLock)
+                {
+                    if (block.Header.HashPrevBlock != this.Tip.HashBlock)
+                    {
+                        this.logger.LogError("(-)[BLOCKMINED_INVALID_PREVIOUS_TIP]");
```

Comment:
`]"` -> `]:null"` In order to have return value in the log

Also exit logs should not be on error level, they should be on trace level. Also note that it is perfectly possible and normal situation that we will mine block just after consensus advanced. So this is not even an error condition, we don't need to inform users of the console about it

Reply:
Fixed 👍